src/executables/sysrepo-plugind.c

```diff
@@ -44,6 +44,11 @@ int loop_finish;
 pthread_mutex_t lock = PTHREAD_MUTEX_INITIALIZER;
 pthread_cond_t cond = PTHREAD_COND_INITIALIZER;
 
+/** Variables for PID file handling */
+int pidfd = -1;
+char *pidfile = NULL;
+char *pid = NULL;
+
```

Comment:
Never use global variables unless necessary. They should be local just like before and passed as parameters to the functions.